### ISSUE TITLE
[SCREENREADER]: Alert Boxes - Role "alert" MUST announce itself to assistive tech on page load, and on change

### ISSUE THREAD

**Issue by 1Copenut** [department-of-veterans-affairs/va.gov-team#10478]
# [508-defect-2](https://github.com/department-of-veterans-affairs/va.gov-team/blob/master/platform/accessibility/guidance/defect-severity-rubric.md#508-defect-2)

**Feedback framework**

- **❗️ Must** for if the feedback must be applied 
- **⚠️ Should** if the feedback is best practice 
- **✔️ Consider** for suggestions/enhancements 

## Description

Our [Alertbox error variant](https://design.va.gov/components/alertboxes#error-alert) of the Alertbox component contains a `role="alert"` for announcing important content to screen readers immediately. In testing, I discovered that several screen reader + browser combinations do not announce these alerts when they are rendered on first page load:

* Win10 + Firefox + NVDA
* Win10 + IE11 + JAWS
* iOS + VoiceOver + Mobile Safari

We need to come up with a strategy to inject the `role="alert"` after the component is rendered or another way to trigger a change in the component so this type of alert is read out regardless of when it is added to the document object model (DOM).

## Point of Contact

**VFS Point of Contact:** _Trevor_

## Acceptance Criteria

- [ ] Error alerts are read out on first page render
- [ ] Error alerts are read out when they are dynamically added to a page

## Environment

* https://design.va.gov/components/alertboxes#error-alert

## WCAG or Vendor Guidance (optional)

* [Name, Role, Value: Understanding SC 4.1.2](https://www.w3.org/TR/UNDERSTANDING-WCAG20/ensure-compat-rsv.html)

## Screenshots or Trace Logs

![Screen Shot 2020-06-19 at 12 21 55 PM](https://user-images.githubusercontent.com/934879/85162917-b34f2a80-b227-11ea-8bac-f86429c10e71.png)

**Comment by 1Copenut** [comment]
Issue moved to [department-of-veterans-affairs/vets-design-system-documentation #204](https://github.com/department-of-veterans-affairs/vets-design-system-documentation/issues/204) via [**ZenHub**](https://www.zenhub.com/)
